src/test/crimson/seastore/test_object_data_handler.cc

```diff
@@ -671,6 +678,34 @@ TEST_P(object_data_handler_test_t, multiple_overwrite) {
   });
 }
 
+TEST_P(object_data_handler_test_t, random_overwrite) {
+  constexpr size_t TOTAL = 4<<20;
+  constexpr size_t BSIZE = 4<<10;
+  constexpr size_t BLOCKS = TOTAL / BSIZE;
+  run_async([this] {
+    set_overwrite_threshold();
+    for (unsigned i = 0; i < BLOCKS; ++i) {
+      auto t = create_mutate_transaction();
+      write(i * BSIZE, BSIZE, 'a');
+    }
+
+    for (unsigned i = 0; i < 4; ++i) {
+      for (unsigned j = 0; j < 100; ++j) {
+	auto t = create_mutate_transaction();
+	for (unsigned k = 0; k < 2; ++k) {
+	  write(*t, get_random_laddr(BSIZE, TOTAL), BSIZE,
```

Comment:
Suggest to mutate the data smaller than the original size.

Is it guaranteed that trim-dirty will always be triggered ?

Reply:
Added and set `target_dirty_bytes` to  `roll_size / 36` to always trigger trim-dirty.